DxDispatch/src/dxdispatch/Device.cpp

```diff
@@ -315,6 +322,20 @@ void Device::DispatchAndWait()
     m_temporaryResources.clear();
 }
 
+void Device::DispatchDontWait()
```

Comment:
Drawing attention to this function, is this the correct subset of calls from DispatchAndWait()? I will delete the commented lines before committing (I left them for context).

Reply:
Almost! You can delete the commented lines, but also delete the last line that clears temporary resources. D3D resources need to be kept alive until the GPU has completed the commands that reference those resources. This is safe to do in DispatchAndWait because of the GPU/CPU sync (commands must have completed on the GPU). However, you can't be sure of this in DispatchDontWait.

Reply:
ALmost! You also need to delete the line that releases temporary buffers. It's only safe to delete temporaries when the commands reference those resources have finished on the GPU, which is unlikely without the CPU/GPU sync.